meson: Missing sys/vfs.h header when compiling libmount on macOS

By default, Meson attempts to build libmount so long as `libblkid` is enabled. On macOS, this doesn't work because `libmount` requires the header `sys/vfs.h` which does not exist on macOS. See the following error message. Meson should probably not attempt to build `libmount` if `sys/vfs.h` is not available on the system.

```
[129/276] Compiling C object libmount/lib_mount.a.p/src_init.c.o
FAILED: libmount/lib_mount.a.p/src_init.c.o 
clang -Ilibmount/lib_mount.a.p -Ilibmount -I../src/libmount -Iinclude -I../src/include -I../src/libmount/src -Ilibblkid -I../src/libblkid -fdiagnostics-color=always -Wall -Winvalid-pch -std=c99 -O0 -g -D_GNU_SOURCE -include /Users/jenkins/workspace/prod-v1/bsr/18274/fcbdf/.conan/data/util-linux-libuuid/2.40/_/_/build/4d92cd78cd137a8eee120fd1d9bc351bd91a44b6/build-debug/config.h -fno-common -Waddress-of-packed-member -Wembedded-directive -Wextra-semi -Wformat-security -Wimplicit-function-declaration -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wno-missing-field-initializers -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wuninitialized -Wunused-parameter -Wunused-result -Wunused-variable -Wvla -Walloca -isysroot /Applications/conan/xcode/13.0/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -arch arm64 -MD -MQ libmount/lib_mount.a.p/src_init.c.o -MF libmount/lib_mount.a.p/src_init.c.o.d -o libmount/lib_mount.a.p/src_init.c.o -c ../src/libmount/src/init.c
In file included from ../src/libmount/src/init.c:21:
../src/libmount/src/mountP.h:22:10: fatal error: 'sys/vfs.h' file not found
#include <sys/vfs.h>
         ^~~~~~~~~~~
1 error generated.
```

--- a/meson.build
+++ b/meson.build
@@ -102,7 +102,12 @@ conf.set('HAVE_MOUNTFD_API', have_mountfd_api ? 1 : false)
 have_struct_statx = cc.sizeof('struct statx', prefix : '#include <sys/stat.h>') > 0
 conf.set('HAVE_STRUCT_STATX', have_struct_statx ? 1 : false)
 
-build_libmount = get_option('build-libmount').require(get_option('build-libblkid').allowed()).allowed()
+have_sys_vfs_header = cc.has_header('sys/vfs.h')
+
+build_libmount = get_option('build-libmount').require(
+  get_option('build-libblkid').allowed() \
+  and have_sys_vfs_header \
+).allowed()
 
 conf.set('HAVE_LIBMOUNT', build_libmount ? 1 : false)
 conf.set('USE_LIBMOUNT_SUPPORT_NAMESPACES', 1)
@@ -1501,7 +1506,7 @@ if opt and not is_disabler(exe)
   bashcompletions += ['tunelp']
 endif
 
-opt = not get_option('build-fstrim').disabled()
+opt = get_option('build-fstrim').require(have_sys_vfs_header).allowed()
 exe = executable(
   'fstrim',
   fstrim_sources,

